### PR TITLE
Make "Using computed attribute values" normative

### LINKED ISSUE #249: Non-Normative Section 6.4 contains normative keyword
>   Implementations MUST compute attribute values based on the contents of the 
document instance before applying those computed values to  DAPT data model objects.

This may be derived by TTML2 but it is still not defined as such in TTML2.

### PR DESCRIPTION
**Pull request by nigelmegitt**
Closes #249.

Retain and reformat the example about non-DAPT functionality.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/dapt/pull/256.html" title="Last updated on Nov 13, 2024, 3:18 PM UTC (d0e3e3e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/dapt/256/e098bac...d0e3e3e.html" title="Last updated on Nov 13, 2024, 3:18 PM UTC (d0e3e3e)">Diff</a>

### CODE REVIEW

**Review by andreastai — APPROVED** [review]
Looks good to me! Thanks!

File: index.html [hunk 1]
```diff
@@ -2554,12 +2553,13 @@ <h3>Using computed attribute values</h3>
           of its descendant <code>&lt;div&gt;</code> elements that do correspond to <a>Script Events</a>
           are different to what they would be if that intermediate <code>&lt;div&gt;</code> element's <code>begin</code>
```
Comment (cconcolato) [inline index.html:2554]: Reading this text again, I find the parts about "that element does not correspond to a Script Event" confusing. It shouldn't matter whether or not it is a Script Event. Inheritance applies. Also it gives the impression that under some circumstances, intermediate div can be ignored. 
I would remove the whole sentence starting with "If". Otherwise, I would rephrase saying something like: 
> If a workflow decides to rewrite a DAPT document to remove nested div elements, care should be taken that if those elements had inheritable attributes like begin, the attribute values like begin on the nested divs should be adjusted.

File: index.html [hunk 2]
```diff
@@ -2554,12 +2553,13 @@ <h3>Using computed attribute values</h3>
...
           time were ignored.</p>
-        <p class="ednote">MAYBE REMOVE Or, also for example, if styling is used in a TTML document, the computed value of the <code>tts:fontSize</code>
+        <aside class="example">An example of this relating to functionality not required by DAPT is styling.
+          If styling is used in a TTML document, the computed value of the <code>tts:fontSize</code>
           attribute when expressed as a percentage is relative to the computed value of the same attribute
           on the element's parent.
           Furthermore, in this example, the algorithm for generating ISDs (Intermediate Synchronic Documents)
           involves reparenting some elements. The computed value that applies is the one <em>before</em>
-          rehoming has taken place.</p>
+          rehoming has taken place.</aside>
```
Comment (cconcolato) [inline index.html:2562]: Do we really need this example? It is about a "functionality not required by DAPT". I would remove it altogether. If not, I would rephrase saying something like:
> If a DAPT document contains additional features not required by DAPT such as styling and the DAPT document is processed by a processor supporting these features, support for inheritance of the inheritable features such as tts:fontSize, is required, per TTML2.

File: index.html (hunk 1 above)
Comment (nigelmegitt) [reply index.html:2554]: I don't like the idea of making this example specifically about rewriting documents - it's about the correct interpretation of a document, regardless of what is being done with it.

Possibly this comment should be raised as a separate issue?

I'd be happy to emphasise further (informatively) that ignoring the intermediate `<div>` is not an acceptable thing to do.

File: index.html (hunk 2 above)
Comment (nigelmegitt) [reply index.html:2562]: Removing it was one of the options before, but until nobody came down on the side of "remove". Not wanting to leave an editorial note in, if avoidable, I just turned it into an example. I don't mind deleting it if it isn't helpful.

File: index.html (hunk 1 above)
Comment (cconcolato) [reply index.html:2561]: My suggestion was a tentative rewrite of what I thought you were trying say. The alternative was to remove the sentence. My main problem is the current emphasis on elements that do not correspond to Script Events. We should not encourage that.

File: index.html (hunk 2 above)
Comment (cconcolato) [reply index.html:2569]: Is my rephrasing not acceptable?

Comment (nigelmegitt) [reply index.html:2569]: Sorry, I understood your preference to be for removal, and didn't focus on the rephrasing. Reading it again, I think the following two paragraphs already say the same thing as the rephrased version - maybe we should just add to one of them the example of `tts:fontSize` to make the abstract statements more concrete, and remove the example?

Concretely, I'm suggesting the paragraph:

>The semantics defined by [[TTML2](https://pr-preview.s3.amazonaws.com/w3c/dapt/pull/256.html#bib-ttml2)] or, for vocabulary defined herein, this specification, take precedence in this scenario. Implementations MUST compute attribute values based on the contents of the [document instance](https://www.w3.org/TR/ttml2/#terms-document-instance) before applying those computed values to DAPT data model objects.

would become:

>The semantics defined by [[TTML2](https://pr-preview.s3.amazonaws.com/w3c/dapt/pull/256.html#bib-ttml2)] or, for vocabulary defined herein, this specification, take precedence in this scenario. Implementations MUST compute attribute values based on the contents of the [document instance](https://www.w3.org/TR/ttml2/#terms-document-instance) before applying those computed values to DAPT data model objects. **For example a processor that supports TTML2 styling features would need to implement the TTML2 semantics for inheritance and computing relative values of attributes like `tts:fontSize`.**

Comment (cconcolato) [reply index.html:2569]: and remove the example?

Comment (nigelmegitt) [reply index.html:2569]: Yes, and remove the example.

File: index.html (hunk 1 above)
Comment (nigelmegitt) [reply index.html:2561]: Here's another attempt at what I was trying to say, maybe this is less confusing?

>Or, for another example, the computed times of an element in a DAPT document are relative to the begin time of its parent element. If a `<div>` element specifies a `begin` attribute, then the computed times of its child `<div>` elements are relative to that intermediate `<div>` element's begin time, and so on down the hierarchy. It is important to include those intermediate `<div>` elements' times in the computation even if the processing target is an instance of the DAPT data model in which they have no direct equivalent; otherwise the [Script Event](https://www.w3.org/TR/dapt/#dfn-script-event) Begin and End times would be wrong.

File: index.html (hunk 2 above)
Comment (nigelmegitt) [reply index.html]: Done in 5617ab0.

File: index.html (hunk 1 above)
Comment (nigelmegitt) [reply index.html]: Change pushed in 33c33fe. On re-reading, I made a couple of small tweaks because I thought the concept of an "intermediate" div wasn't really explained properly.

**Review by cconcolato — APPROVED** [review]
Overall, I'm approving the PR to unlock the CR. 

That said, I don't think it is a substantive change because the changes are about examples. The section contains a single MUST statement that is not changed.

Anyway, I think we could improve the section editorially later on. I find it scary for implementers because the boundaries of the implementation requirements are not clear. There are many examples (xml:lang, begin with nested divs, styles), but the concrete normative text is not visible enough, upfront. As I understand it, there are only 2 concrete consequences on implementers:
* use of style TTML2 properties, and there are 2 cases: either the implementation cares about them and it needs to implement proper inheritance; or it does not care about them and can fully ignore them
* use of nested divs (which I'm not a big fan of...). Implementations don't have a choice. To be compliant, they have to apply the inheritance model on intermediate divs.

**Comment by nigelmegitt** [comment]
> That said, I don't think it is a substantive change because the changes are about examples. The section contains a single MUST statement that is not changed.

Thanks @cconcolato - the reason the section was made normative was because it had a MUST statement in, but was previously marked as non-normative.

> there are only 2 concrete consequences on implementers

Not just style properties and nested divs, also timing properties and anything else that can be put on a div, like `xml:lang`.

Thanks for the approval, I will merge, and if more problems or improvements are identified they can be dealt with in new issues and pull requests.
